### PR TITLE
Thermistor Support

### LINKED ISSUE #39: Thermistor Support
Given:
A thermistor sensor is connected to the Raspberry Pi.

When:
The temperature is measured.

Then:
Real-time temperature data is provided in Celsius or Fahrenheit.

**Component Functionality:**
This component will enable the Pi4Micronaut project to measure temperature through a thermistor, providing real-time temperature data for monitoring and automation tasks.

**Developer Notes:**

1. Draw the circuit setup for hardware connections: Show how the thermistor connects to the Raspberry Pi via an analog-to-digital converter (ADC).
2.  Develop functionality to read temperature: Create a ThermistorHelper.java class to read raw data from the thermistor and convert it to temperature in Celsius or Fahrenheit.
3.  Implement conversion algorithms: Convert the resistance reading from the thermistor into a meaningful temperature value using the appropriate formula.
4. Configure GPIO pins and ADC appropriately: Ensure compatibility with various thermistors and ADCs.
5. Create an example controller in componets folder to clearly demonstate the working of helper class.
6.  Create a YAML configuration file for sensor settings: Add fields like name, address, pull_Down, debounce, provider.
7. Provide code comments for each function for future documentation.

Acceptance Criteria:

- [ ] Functionality to read the temperature data: The thermistor should provide accurate temperature readings.
- [ ] Component defined in YAML file with:
    - [ ] name
    - [ ] address
    - [ ] pull
    - [ ] debounce
    - [ ] provider
- [ ] Schematic diagram accurately depicts the wiring for the thermistor and ADC.
- [ ] Controller Implementation in components folder
- [ ] Code comments provided for each functionality for documentation purposes.

### PR DESCRIPTION
**Pull request by ThomasMacas**
Fixes #39 

In order to integrate a thermistor as another component a Controller and Helper files are needed. In order for the Raspberry pi to integrate with our software the Application.yml file was adjusted to accommodate for the thermistor. The ThermistorHelper.java file was create to give the basic operations for the thermistor and the ThermistorController.java as a basic implementation of it.

### CODE REVIEW

File: components/src/main/resources/application.yml
Comment (leandrumartin) [inline components/src/main/resources/application.yml:1]: Should there be a `thermistor-output` under `digital-output`? It looks like the controller takes one in as a constructor parameter.

File: components/src/main/java/com/opensourcewithslu/components/controllers/ThermistorController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.pi4j.io.gpio.digital.DigitalOutput;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import jakarta.inject.Named;
+
+@Controller("/thermistor")
+public class ThermistorController {
+    private final ThermistorHelper thermistorHelper;
+    // Finalize (Still gathering my thoughts about this)
+    public ThermistorController(@Named("thermistor-input") DigitalInput sensorInput, @Named("thermistor-output") DigitalOutput sensorOutput){
```
Comment (leandrumartin) [inline components/src/main/java/com/opensourcewithslu/components/controllers/ThermistorController.java:14]: If the only endpoint is to read a value, do we need a DigitalOutput here? Or are there more endpoints planned that involve using the thermistor as an output for something?

**Review by leandrumartin — CHANGES_REQUESTED** [review]

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
There is the actual functionality to be implemented in the helper class. Please ensure all planned features are fully developed and tested with actual hardware to confirm correct communication protocols. Once everything is finalized, we’ll be ready for a thorough review.

**Review by yrlmanoharreddy — APPROVED** [review]
This pull request implements support for the thermistor component; however, we currently lack the necessary Analog-to-Digital Converter (ADC) to test the hardware functionality effectively. To ensure thorough testing, I plan to prioritize ADC support in the next sprint and test the thermistor hardware in parallel.

**Pull request merged**